lib/nat-analyzer.js

```diff
@@ -0,0 +1,117 @@
+// how far can the port median distance be?
```

Comment:
should nat-analyzer be it's own module? will there be any future use cases apart from dht-rpc?

Reply:
(my thinking on this is about isolating the tests for it as well)

Reply:
Yea, that's not a bad idea. Just a new module that only really useful here :+1:

Reply:
Gonna move it out after releasing the rc